src/grt/src/GlobalRouter.cpp

```diff
@@ -498,6 +499,28 @@ void GlobalRouter::initRoutingLayers()
   fastroute_->setLayerOrientation(vertical);
 }
 
+void GlobalRouter::checkAdjacentLayersDirection(int min_routing_layer,
+                                                int max_routing_layer)
+{
+  odb::dbTech* tech = db_->getTech();
+  for (int l = min_routing_layer; l < max_routing_layer; l++) {
+    odb::dbTechLayer* layer_a = tech->findRoutingLayer(l);
+    odb::dbTechLayer* layer_b = tech->findRoutingLayer(l + 1);
```

Comment:
should the loop end at max_routing_layer - 1 ?

Reply:
No. The range is [min_routing_layer, max_routing_layer], so when `l` is `max_routing_layer - 1`, it will get the correct layers without going out of the range.